src/CattyTests/Scene/CBSpriteNodePenExtensionTests.swift

```diff
@@ -0,0 +1,147 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+final class CBSpriteNodePenExtensionTests: XCTestCase {
+
+    var scene: CBScene!
+    var spriteNode: CBSpriteNode!
+    var initialPosition: CGPoint!
+
+    override func setUp() {
+        spriteNode = CBSpriteNode(spriteObject: SpriteObject())
+        spriteNode.name = "testName"
+
+        scene = SceneBuilder(project: ProjectMock()).build()
+        scene.addChild(spriteNode)
+
+        initialPosition = CGPoint.zero
+    }
+
+    func testWhenSpritePositionChangedPenUp() {
+        spriteNode.penConfiguration.penDown = false
+        spriteNode.penConfiguration.previousPosition = initialPosition
+        spriteNode.position = CGPoint(x: 1, y: 1)
+
+        XCTAssertEqual(scene.children.count, 1)
+
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 1)
+    }
+
+    func testWhenSpritePositionChangedPenDown() {
+        spriteNode.penConfiguration.penDown = true
+        spriteNode.position = initialPosition
+
+        XCTAssertEqual(scene.children.count, 1)
+
+        spriteNode.update(CACurrentMediaTime())
+        spriteNode.position = CGPoint(x: 1, y: 1)
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 2)
+
+        var allShapeNodes = [LineShapeNode]()
+
+        scene.enumerateChildNodes(withName: SpriteKitDefines.penShapeNodeName) { node, _ in
```

Comment:
Please create a private method to avoid code duplications (e.g., below the `setUp`).